client_handlers.go

```diff
@@ -45,9 +47,26 @@ func clientHandshakeHandler(c *Conn) error {
 		}
 
 		if c.localPSKCallback == nil {
-			expectedHash := valueKeySignature(clientRandom, serverRandom, h.publicKey, h.namedCurve, h.hashAlgorithm)
-			if err := verifyKeySignature(expectedHash, h.signature, h.hashAlgorithm, c.state.remoteCertificate); err != nil {
-				return err
+			if !c.insecureSkipVerify {
+				expectedHash := valueKeySignature(clientRandom, serverRandom, h.publicKey, h.namedCurve, h.hashAlgorithm)
+				if err := verifyKeySignature(expectedHash, h.signature, h.hashAlgorithm, c.state.remoteCertificate); err != nil {
+					return err
+				}
+				opts := x509.VerifyOptions{
```

Comment:
I would move the actual verification into `crypto.go`, just so it is one easy place to check out

Reply:
Done. I've created `verifyClientCert` and `verifyServerCert` at `crypto.go`.